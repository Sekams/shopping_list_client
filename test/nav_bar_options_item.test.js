import React from 'react';
import NavBarOptionItem from '../src/components/nav_bar_option_item'

test("renders the nav_bar_option_item component", () => {
    const wrapper = shallow(
        <NavBarOptionItem />
    );
    expect(wrapper).toMatchSnapshot();
});

test("renders the nav bar option items for the home page", () => {
    const props = {
        logged_in: true,
        authorized: true,
        page: 'home',
        name: 'search',
        onSearchTermChange: jest.fn(),
    };
    const wrapper = shallow(
        <NavBarOptionItem {...props} />
    );
    wrapper.find('a').map((anchor) => {
        anchor.simulate('click', { preventDefault() { } });
    });
    wrapper.props().children[0].props.children.props.onSearchTermChange();
    wrapper.props().children[1].props.children[1].props.setPageLimit();
    wrapper.props().children[1].props.children[1].props.showModal({ preventDefault() { } })
    expect(wrapper).toMatchSnapshot();
});

test("renders the nav bar option items for the home page greeting", () => {
    const props = {
        logged_in: true,
        authorized: true,
        page: 'home',
        name: 'greeting',
        username: 'atom',
    };
    const wrapper = shallow(
        <NavBarOptionItem {...props} />
    );
    wrapper.instance().showModal({ preventDefault() { } });
    wrapper.update();
    wrapper.find('a').map((anchor) => {
        anchor.simulate('click', { preventDefault() { } });
    });
    expect(wrapper).toMatchSnapshot();
});

test("renders the nav bar option items for the home page logout", () => {
    const props = {
        logged_in: true,
        authorized: true,
        page: 'home',
        name: 'link',
    };
    const wrapper = shallow(
        <NavBarOptionItem {...props} />
    );
    wrapper.find('a').map((anchor) => {
        anchor.simulate('click', { preventDefault() { } });
    });
    expect(wrapper).toMatchSnapshot();
});

test("renders the nav bar option items for the sign up page", () => {
    const props = {
        logged_in: true,
        authorized: true,
        page: 'sign_up',
        name: 'link',
    };
    const wrapper = shallow(
        <NavBarOptionItem {...props} />
    );
    expect(wrapper).toMatchSnapshot();
});

test("renders the nav bar option items for the login page", () => {
    const props = {
        logged_in: true,
        authorized: true,
        page: 'login',
        name: 'link',
    };
    const wrapper = shallow(
        <NavBarOptionItem {...props} />
    );
    expect(wrapper).toMatchSnapshot();
});

test("renders the nav bar option items for the home page logout", () => {
    fetch.mockResponseOnce(JSON.stringify(
        {
            "message": "You successfully logged out.",
            "status": "success",
        }),
        {
            status: 200,
            ok: true,
        }
    );
    const props = {
        logged_in: true,
        authorized: true,
        page: 'home',
        name: 'link',
    };
    const wrapper = shallow(
        <NavBarOptionItem {...props} />
    );
    wrapper.find('a').map((anchor) => {
        anchor.simulate('click', { preventDefault() { } });
    });
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
});