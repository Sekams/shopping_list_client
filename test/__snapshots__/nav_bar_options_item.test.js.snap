// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles missing access token 1`] = `""`;

exports[`handles toggle dropdown 1`] = `""`;

exports[`renders the nav bar option items for the home page 1`] = `
<li>
  <div
    className="navbar-input-divs"
  >
    <SearchBar
      onSearchTermChange={[Function]}
    />
  </div>
  <div
    className="navbar-input-divs navbar-serch-limit"
  >
    <h5
      className="nav-text"
    >
      <a
        className="navbar-link"
        href=""
        onClick={[Function]}
      >
        <b
          className="caret"
        />
      </a>
       Lists per page
    </h5>
    <NavBarDropdown
      owner="page_limit"
      setPageLimit={[Function]}
      showModal={[Function]}
    />
  </div>
</li>
`;

exports[`renders the nav bar option items for the home page greeting 1`] = `
<li>
  <h5
    className="nav-text"
  >
    Hi 
    <a
      className="navbar-link"
      href=""
      onClick={[Function]}
    >
      atom
      <b
        className="caret"
      />
    </a>
  </h5>
  <NavBarDropdown
    owner="change_password"
    setPageLimit={[Function]}
    showModal={[Function]}
  />
</li>
`;

exports[`renders the nav bar option items for the home page logout 1`] = `
<li>
  <a
    className="navbar-link"
    href=""
    onClick={[Function]}
  >
    Logout
  </a>
</li>
`;

exports[`renders the nav bar option items for the home page logout 2`] = `
<li>
  <a
    className="navbar-link"
    href=""
    onClick={[Function]}
  >
    Logout
  </a>
</li>
`;

exports[`renders the nav bar option items for the login page 1`] = `
<li>
  <Link
    className="navbar-link"
    replace={false}
    to="/register"
  >
    Sign Up
  </Link>
</li>
`;

exports[`renders the nav bar option items for the sign up page 1`] = `
<li>
  <Link
    className="navbar-link"
    replace={false}
    to="/login"
  >
    Login
  </Link>
</li>
`;

exports[`renders the nav_bar_option_item component 1`] = `""`;
